Applications/Spire/Source/Ui/RegionListItem.cpp

```diff
@@ -0,0 +1,135 @@
+#include "Spire/Ui/RegionListItem.hpp"
+#include <QHBoxLayout>
+#include <QVBoxLayout>
+#include "Nexus/Definitions/DefaultCountryDatabase.hpp"
+#include "Spire/Spire/Dimensions.hpp"
+#include "Spire/Ui/Icon.hpp"
+#include "Spire/Ui/TextBox.hpp"
+
+using namespace boost;
+using namespace Nexus;
+using namespace Spire;
+using namespace Spire::Styles;
+
+namespace {
+  auto ICON_SIZE() {
+    static auto size = scale(10, 10);
+    return size;
+  }
+
+  auto FLAG_SIZE() {
+    static auto size = scale(16, 10);
+    return size;
+  }
+
+  auto ICON_STYLE() {
+    auto style = StyleSheet();
+    style.get(Any()).
+      set(BackgroundColor(QColor::fromRgb(0, 0, 0, 0))).
+      set(Fill(none));
+    return style;
+  }
+
+  auto NAME_LABEL_STYLE(StyleSheet style) {
+    auto font = QFont("Roboto");
+    font.setWeight(QFont::Normal);
+    font.setPixelSize(scale_width(10));
+    style.get(ReadOnly() && Disabled()).
+      set(text_style(font, QColor::fromRgb(0x80, 0x80, 0x80))).
+      set(PaddingBottom(0)).
+      set(PaddingTop(scale_height(4)));
+    return style;
+  }
+
+  auto VALUE_LABEL_STYLE(StyleSheet style) {
+    style.get(ReadOnly() && Disabled()).
+      set(PaddingRight(scale_width(8))).
+      set(vertical_padding(0));
+    return style;
+  }
+}
+
+RegionListItem::RegionListItem(Region region, QWidget* parent)
+    : QWidget(parent),
+      m_region(std::move(region)),
+      m_type(get_type()) {
+  auto value_label = make_value_label();
+  value_label->setSizePolicy(QSizePolicy::Expanding, QSizePolicy::Preferred);
+  set_style(*value_label, VALUE_LABEL_STYLE(get_style(*value_label)));
+  auto name_label = make_label(QString::fromStdString(m_region.GetName()));
+  name_label->setSizePolicy(QSizePolicy::Expanding, QSizePolicy::Preferred);
+  set_style(*name_label, NAME_LABEL_STYLE(get_style(*name_label)));
+  auto type_icon = make_type_icon();
+  auto value_container_layout = new QHBoxLayout();
+  value_container_layout->setContentsMargins({});
+  value_container_layout->setSpacing(0);
+  value_container_layout->addWidget(value_label);
```

Comment:
Even here, there is a jump from value to name back to value. Keep code local. Local reasoning is a lot easier than global reasoning.